packages/cardpay-reward-programs/cardpay_reward_programs/rules/min_spend.py

```diff
@@ -24,10 +24,10 @@ def sql(self, table_query):
         sum(spend_amount_uint64) as total_spent
 
         from {table_query}
-        where block_number_uint64 > ?::integer and block_number_uint64 <= ?::integer 
+        where block_number_uint64 > $1::integer and block_number_uint64 <= $2::integer 
         
         group by prepaid_card_owner
-        having(total_spent) >= ?::integer
+        having(total_spent) >= $3::integer
```

Comment:
Found that duckdb supports numbered parameters - not as nice as named ones but they are not implemented. I think this will be a little easier to keep going.